### PR TITLE
Tech/separate attachment and data store

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java [hunk 1]
```diff
@@ -58,7 +54,7 @@
 @EnableJpaRepositories(basePackages = "org.opentestsystem.ap.common.datastore.repository")
 @ConditionalOnProperty(prefix = "itembank.enabled", name = "database")
 @EnableConfigurationProperties({ItemBankProperties.class})
-public class ItemDataStoreManagerConfiguration {
+public class DataStoreDataManagerConfiguration {
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java]: I misnamed this configuration class.  It is the DataStoreDataManager configuration which deals with items but isn't the ItemDataStoreManager config.

File: common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java [hunk 2]
```diff
@@ -77,27 +73,6 @@ public IdGenerator itemIdGenerator(ItemBankProperties itemBankProperties, Sequen
             sequenceRepository);
     }
 
-    @Bean
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java]: Moved this out into DataStoreAttachmentManagerConfiguration.  Doing this allows services that need to use DataStoreDataManager but don't need to deal with S3 can use this config.

File: common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java [hunk 3]
```diff
@@ -118,16 +93,10 @@ public DataStoreDataManager dataStoreDataManager(ItemBankProperties itemBankProp
             itemAttachmentMapper);
     }
 
-    @Bean
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java]: Ditto for this moving to its own config.

File: common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java [hunk 4]
```diff
@@ -118,16 +93,10 @@ public DataStoreDataManager dataStoreDataManager(ItemBankProperties itemBankProp
...
-    public DataStoreAttachmentManager dataStoreAttachmentManager(final ItemBankProperties itemBankProperties) {
-        log.info("Creating bean dataStoreAttachmentManager");
-        return new DataStoreAttachmentManager(s3BucketManager(itemBankProperties), dataStoreUtility());
-    }
-
     @Bean
     @ConditionalOnMissingBean
     public ItemAttachmentMapper itemAttachmentMapper() {
-        return new ItemAttachmentMapperImpl(Arrays.asList(new GeneralItemAttachmentMapper(), new GiItemAttachmentMapper()));
+        return new ItemAttachmentMapperImpl(
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/DataStoreDataManagerConfiguration.java]: Ignore - formatter change.

File: common/src/main/java/org/opentestsystem/ap/common/config/ItemManagerConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * Configuration class that configures the Item Manager and its dependencies
+ */
+@Slf4j
+@Import({
+    DataStoreUtilityConfiguration.class,
+    ItemAttachmentDataStoreManagerConfiguration.class,
+    DataStoreDataManagerConfiguration.class
+})
+public class ItemManagerConfiguration {
```
Comment (millerts) [inline common/src/main/java/org/opentestsystem/ap/common/config/ItemManagerConfiguration.java]: This configuration was in a bunch of microservices.  I was making changes to the ItemDataStoreManager as part of my migration work, and I was surprised at the number of microservices I touched.  This centralizes the configuration around item management.

File: item-bulk-management-service/src/main/java/org/opentestsystem/ap/ibms/ItemBulkConfiguration.java [hunk 6]
```diff
@@ -189,33 +182,13 @@ public TestAdminProcessor testAdminProcessor() {
...
-            this.dataStoreAttachmentManager,
-            itemManagerEventProducer(),
-            this.dataStoreUtility,
-            this.itemAttachmentMapper);
-    }
-
-    @Bean
-    @ConditionalOnMissingBean
-    public ItemBulkUpdateItemManager itemBulkUpdateItemManager() {
+    public ItemBulkUpdateItemManager itemBulkUpdateItemManager(final DataStoreItemManager dataStoreItemManager) {
```
Comment (millerts) [inline item-bulk-management-service/src/main/java/org/opentestsystem/ap/ibms/ItemBulkConfiguration.java]: You'll see these type of changes throughout the review.  My preference is to pass in the dependencies as part of the bean method which will help future refactoring.  The calling of internal methods in this class results in more changes when teasing out configurations.

File: common/src/main/java/org/opentestsystem/ap/common/config/ItemManagerConfiguration.java (hunk 5 above)
Comment (malaffoon) [reply common/src/main/java/org/opentestsystem/ap/common/config/ItemManagerConfiguration.java]: So now you can touch dozens of microservices without them even knowing!

Comment (wtritch) [reply common/src/main/java/org/opentestsystem/ap/common/config/ItemManagerConfiguration.java]: 'zactly.  I don't care *how* the system gives me a Foob, or what a Foob needs under the covers.  I just want my damn Foob.

**Pull request merged**